features/netsocket/TCPServer.cpp

```diff
@@ -20,10 +20,12 @@ using mbed::Callback;
 
 TCPServer::TCPServer()
 {
+    _socket_stats.stats_new_socket_entry(this);
 }
 
 TCPServer::~TCPServer()
 {
+    _socket_stats.stats_update_socket_state(this, SOCK_CLOSED);
```

Comment:
Move this into the `InternetSocket::close()` because socket can be closed way before it is destructed.